app/src/main/java/cm/aptoide/pt/search/view/SearchResultPresenter.java

```diff
@@ -205,15 +205,24 @@ private int getItemCount(List<SearchAppResult> data) {
         .filter(viewModel -> !viewModel.hasReachedBottomOfFollowedStores())
         .observeOn(viewScheduler)
         .doOnNext(__ -> view.showLoadingMore())
-        .flatMapSingle(viewModel -> loadDataFromFollowedStores(viewModel.getSearchQueryModel()
-            .getFinalQuery(), viewModel.isOnlyTrustedApps(), viewModel.getFollowedStoresOffset()))
+        .flatMapSingle(viewModel -> {
+          String storeName = viewModel.getStoreName();
+          if (storeName != null && !storeName.trim()
+              .equals("")) {
+            return loadDataForSpecificStore(viewModel.getSearchQueryModel()
+                    .getFinalQuery(), storeName, viewModel.getFollowedStoresOffset()).map(
+                    searchResult -> new Pair<>(searchResult, null));
```

Comment:
why did you add this map for this pair ?
This is the reason why you have to cast data to searchresult on line #225

Reply:
No idea why I did that 🚀